### PR TITLE
g.extension: better handle request exceptions for l flag

### PR DESCRIPTION
**Pull request by tmszi**
**Describe the bug**
When you disable internet connection and try download list of available addons via `g.extension -l` module, g.extension will try download module.xml file for GRASS GIS 7.8 version (see URL address in error message), due wrong handled request exception on line 254, before try to dowlnoad this file.

https://github.com/OSGeo/grass/blob/f720e3eba5d1462ad0b37f2b020c15f05870b05d/scripts/g.extension/g.extension.py#L247-L256

**To Reproduce**

1. Disable your internet connection
2. Launch `g.extension -l`
4. See error (request exception is handed correctly here, but URL address is incorrect due another request exception before this (line 254) is wrong handled.

```
GRASS nc_basic_spm_grass7/PERMANENT:~ > g.extension -l
List of available extensions (modules):
ERROR: Download file from
       <https://grass.osgeo.org/addons/grass7/modules.xml> failed. Check
       your internet connection.
```

https://grass.osgeo.org/addons/grass7/modules.xml -> **grass7**

Same for:

```GRASS nc_basic_spm_grass7/PERMANENT:~ > g.extension sentinel
ERROR: Download file from
       <https://api.github.com/repos/OSGeo/grass-addons/git/trees/grass7?recursive=1>,
       failed. Check internet connection.
```

https://api.github.com/repos/OSGeo/grass-addons/git/trees/grass7?recursive=1 -> **grass7**

**Expected behavior**
Correctly handled exception if internet connection is not available inside [get_version_branch()](https://github.com/OSGeo/grass/blob/f720e3eba5d1462ad0b37f2b020c15f05870b05d/scripts/g.extension/g.extension.py#L247-L256) function.

Error message should be:

```
GRASS nc_basic_spm_grass7/PERMANENT:~ > g.extension -l
ERROR: You are not connected to the internet. Check your internet
       connection.
```

```
GRASS nc_basic_spm_grass7/PERMANENT:~ > g.extension sentinel
ERROR: You are not connected to the internet. Check your internet
       connection.
```

**System description:**

- GRASS GIS version: main, realesebranch_7_8, realesebranch_8_0

### CODE REVIEW

**Review by ninsbl — APPROVED** [review]
I did not test locally, but it looks good to me.

File: scripts/g.extension/g.extension.py [hunk 1]
```diff
@@ -388,16 +398,31 @@ def etree_fromfile(filename):
...
+        else:
+            gscript.fatal(
+                _(
+                    "{base_message}, {err}. ".format(
+                        base_message=base_message,
+                        err=err,
+                    ),
+                ),
+            )
     except URLError:
```
Comment (wenzeslaus) [inline scripts/g.extension/g.extension.py:421]: `as error` needed and then use it later.

File: scripts/g.extension/g.extension.py [hunk 2]
```diff
@@ -251,8 +251,18 @@ def get_version_branch(major_version):
     version_branch = f"grass{major_version}"
     try:
         urlrequest.urlopen(f"{GIT_URL}/tree/{version_branch}/src")
+    except HTTPError as err:
+        if err.code == 404:  # Not Found
+            version_branch = "grass{}".format(int(major_version) - 1)
+        else:
+            grass.fatal(_(err))
     except URLError:
```
Comment (wenzeslaus) [inline scripts/g.extension/g.extension.py:259]: `as error` needed and then use it later.

File: scripts/g.extension/g.extension.py [hunk 3]
```diff
@@ -388,16 +398,31 @@ def etree_fromfile(filename):
...
+            gscript.fatal(
+                _(
+                    "{base_message}. File not found on the"
+                    " server.".format(base_message=base_message)
+                ),
+            )
+        else:
+            gscript.fatal(
+                _(
+                    "{base_message}, {err}. ".format(
```
Comment (wenzeslaus) [inline scripts/g.extension/g.extension.py:415]: Why comma here? Probably, neither `,` nor `.` are needed here. It should be `"{base_message}: {err}"`.

File: scripts/g.extension/g.extension.py [hunk 4]
```diff
@@ -251,8 +251,18 @@ def get_version_branch(major_version):
...
+        if err.code == 404:  # Not Found
+            version_branch = "grass{}".format(int(major_version) - 1)
+        else:
+            grass.fatal(_(err))
     except URLError:
-        version_branch = "grass{}".format(int(major_version) - 1)
+        grass.fatal(
+            _(
+                "You are not connected to the internet. Check your "
+                " internet connection."
```
Comment (wenzeslaus) [inline scripts/g.extension/g.extension.py:263]: `your "` and `" internet"`: only one space is needed.

File: scripts/g.extension/g.extension.py [hunk 5]
```diff
@@ -388,16 +398,31 @@ def etree_fromfile(filename):
 
 def etree_fromurl(url):
     """Create XML element tree from a given URL"""
+    base_message = "Download file from <{url}> failed".format(url=url)
```
Comment (wenzeslaus) [inline scripts/g.extension/g.extension.py:401]: Needs to be translatable too if you want that to be translatable (I assume yes).

**Review by wenzeslaus — CHANGES_REQUESTED** [review]
Given the errors we see elsewhere, I think adding the original error message to the message is needed (`our text: {err}`) to provide as much details as possible (or to not assume too much when we don't know the details).

**Comment by neteler** [comment]
@tmszi Would you mind to address the comments? Most look like easy to fix.

**Comment by wenzeslaus** [comment]
Current error message for g.extesnion is:

```
$ grass --tmp-location XY --exec g.extension -l
ERROR: Failed to get branch from the Git repository
       <https://github.com/OSGeo/grass-addons/>.
       fatal: unable to access 'https://github.com/OSGeo/grass-addons/':
       Could not resolve host: github.com
```

This does not include "You are not connected to the internet. Check your internet connection." like the suggested message. However, the handling is done in a different way which is what this PR was mostly doing.

The code on main changed completely and the review comments here don't make any sense anymore. I suggest closing this PR.

**Comment by wenzeslaus** [comment]
As I mentioned in my previous comment, this is resolved by other code changes.

Follow with some separate message cosmetics PR if needed.

**Pull request closed without merging**